tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java

```diff
@@ -163,6 +164,62 @@ public List<Class<? extends IBizlogic>> exclusions() {
 
     }
 
+    @Test
+    public void testThatExclusionIsHonoredWhenControlDependencyIsExcluded() throws Exception {
+
+        CapabilityDefinition capabilityDefinition = new EmptyCapabilityDefinition() {
+            @Override
+            public String name() {
+                return "C1";
+            }
+
+            @Override
+            public List<? extends CapabilityDefinition> dependentCapabilities() {
+                return Collections.emptyList();
+            }
+
+            @Override
+            public List<Class<? extends BasicValidationBizlogic>> validators() {
+                return ImmutableList.of(BasicValidationBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends BasicEnrichmentBizlogic>> enrichers() {
+                return ImmutableList.of(BasicEnrichmentBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends DataAdapterBizlogic>> adapters() {
+                return ImmutableList.of(DataAdapterBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends IBizlogic>> exclusions() {
+                return ImmutableList.of(DataAdapterBizlogic2.class);
```

Comment:
Given adapter2 was never added to the flow from any place, this exclusion does not test things properly. I think you should add `DataAdapterBizlogic2` in `adapters`

Reply:
aded